### PR TITLE
EVAKA-4018: 5 year old absences

### PR DESCRIPTION
**Pull request by Joosakur**

### CODE REVIEW

File: frontend/packages/employee-frontend/src/types/absence.ts [hunk 1]
```diff
@@ -53,11 +53,18 @@ export const billableCareTypes: CareType[] = ['PRESCHOOL_DAYCARE', 'DAYCARE']
...
   careType: CareType,
   placementType: PlacementType,
+  dateOfBirth: LocalDate,
   i18n: Translations
 ) {
   const isPreparatory =
     placementType === 'PREPARATORY' || placementType === 'PREPARATORY_DAYCARE'
 
+  const hasFree5YearsOldDaycare =
+    dateOfBirth.year === LocalDate.today().year - 5
```
Comment (jaahon) [inline frontend/packages/employee-frontend/src/types/absence.ts:63]: This is not how it's determined. Children that turn 5 in year xxxx get the discount during the "season" of 1.8.xxxx - 31.7.(xxxx + 1), so basically for the year before preschool.

Comment (Joosakur) [reply frontend/packages/employee-frontend/src/types/absence.ts:63]: Oh.. argh

File: service/src/main/kotlin/fi/espoo/evaka/attendance/ChildAttendanceController.kt [hunk 2]
```diff
@@ -443,3 +436,17 @@ fun wasAbsentFromPreschoolDaycare(placementType: PlacementType, arrived: LocalTi
 
     return false
 }
+
+fun wasAbsentFromPaidDaycare(placementBasics: ChildPlacementBasics, arrived: LocalTime, departed: LocalTime): Boolean {
+    val (placementType, dateOfBirth) = placementBasics
+
+    if (!isEntitledToFreeFiveYearsOldDaycare(dateOfBirth)) {
+        return false
```
Comment (jaahon) [inline service/src/main/kotlin/fi/espoo/evaka/attendance/ChildAttendanceController.kt:444]: Shouldn't this be `true` or am I missing something? I don't fully understand what's going on here.

**Review by jaahon — APPROVED** [review]

File: service/src/main/kotlin/fi/espoo/evaka/attendance/ChildAttendanceController.kt (hunk 2 above)
Comment (Joosakur) [reply service/src/main/kotlin/fi/espoo/evaka/attendance/ChildAttendanceController.kt:444]: No, but the function name may still be misleading. The function is used when child is leaving to determine if we should add an absence for the paid portion of the daycare. That scenario only applies to 5yo case, hence false. Maybe I'll rename it as `wasAbsentFrom5yoPaidDaycare`?

Comment (jaahon) [reply service/src/main/kotlin/fi/espoo/evaka/attendance/ChildAttendanceController.kt:444]: Yeah that rename sounds like a good idea.

**Pull request merged**